STM32F405RGTx/Core/Src/rtos_threads/ltc6813_thread.cpp

```diff
@@ -0,0 +1,29 @@
+#include "ltc6813_thread.hpp"
+#include "ltc6813.h"
+#include "bms_entry.hpp"
+#include "threads.hpp"
+#include "main.h"
+
+RTOSThread Ltc6813Thread::thread;
+
+void Ltc6813Thread::initialize() {
+    thread = RTOSThread(
+        "ltc6813_thread",
+        1024*2,
+        osPriorityBelowNormal,
+        runDriver
+    );
+}
+
+void Ltc6813Thread::runDriver(void* args) {
+    while (1) {
+        Ltc6813_wakeup_sleep(&ltc6813);
+        Ltc6813_wakeup_idle(&ltc6813);
+        if (Ltc6813_read_adc(&ltc6813, NORMAL_ADC)) {
+            for (uint8_t i = 0; i < NUM_CELLS; i++) {
+                global_bms_data.battery.cells[i].voltage = ltc6813.cell_voltages[i];
+            }
+        }
+        osDelay(LTC_MEASUREMENT_PERIODICITY*1E3);
```

Comment:
I know that I was the one who originally did this in the measurements thread, but I actually don't like this lmaoo.

I think the reason why I did that was so the coulomb counting thread could have SI units and not have to do the conversion, but here it's probably better to have this in milliseconds.

On this topic, I think we should probably have **ALL** the thread periodicities in `threads.hpp`, in milliseconds, so it will be easier to parse a "determining schedulability" point of view. The coulomb counting thread can define its own macro to get the periodicity in SI units.

Could u implement that maybe? 👉 👈

Reply:
okie , will implement